Cooper.DAO/ChatDAO/ChatDAO.cs

```diff
@@ -0,0 +1,220 @@
+        private string idColumn;
+        private HashSet<string> attributes;
+
+        public ChatDAO(IConfigProvider configProvider)
+        {
+            crud = new CRUD(configProvider);
+            logger = LogManager.GetCurrentClassLogger();
+            dbConnect =  new DbConnect(configProvider);
+
+            table = "CHATS";
+            idColumn = "ID";
+            attributes = new HashSet<string>()
+            {
+                "ID", "CHATNAME", "PHOTOURL"
+            };
+        }
+        
+        public IList<ChatDb> GetPersonalChats(long userId)
+        {
+            IList<ChatDb> personalChats = null;
+
+            var chatAttributes = new HashSet<string>()
+            {
+                "ID"
+            };
+
+            string sqlExpression = String.Format("SELECT IDCHAT as ID FROM {0} INNER JOIN USERSCHATS ON {0}.ID = USERSCHATS.IDCHAT WHERE IDUSER = {1}", table, userId);
+
+            IList<EntityORM> chatsEntities = ExecuteQuery(chatAttributes, sqlExpression);
+
+            if (chatsEntities != null && chatsEntities.Count != 0)
+            {
+                personalChats = new List<ChatDb>(capacity: chatsEntities.Count);
+                var chatsIndexes = new Dictionary<long, int>(capacity: chatsEntities.Count);    // helping collection
+
+                for (int i = 0; i  < chatsEntities.Count; i++)
+                {
+                    ChatDb chat;
+                    Mapping.EntityMapping.Map(chatsEntities[i], out chat);
+                    chat.Participants = new List<long>();
+
+                    personalChats.Add(chat);
+
+                    chatsIndexes.Add(chat.Id, i);
+                }
+
+                var user_attribute = new HashSet<string> { "IDUSER", "IDCHAT" };
+                sqlExpression = "SELECT IDUSER, IDCHAT FROM USERSCHATS ";
+
+                for (int i = 0; i < personalChats.Count; i++)
+                {
+                    sqlExpression += (i == 0) ? "WHERE " : "OR ";
+                    sqlExpression += $"IDCHAT = {personalChats[i].Id} ";
+                }
+
+                IList<EntityORM> participantsEntities = ExecuteQuery(user_attribute, sqlExpression);
+
+                if (participantsEntities != null)
+                {
+
+                    foreach (var participantEntity in participantsEntities)
+                    {
+                        long participantId = Convert.ToInt64(participantEntity.attributeValue["IDUSER"]);
+                        long chatId = Convert.ToInt64(participantEntity.attributeValue["IDCHAT"]);
+
+                        personalChats[chatsIndexes[chatId]].Participants.Add(participantId);
+                    }
+                }
+            }
+
+            return personalChats;
+
+        }
+
+        public ChatDb GetOnetoOneChatByParticipantsId(IList<long> participantsId)
+        {
+            ChatDb chat = null;
+
+            string sqlExpression = "SELECT chat.id, chat.chatname, chat.photourl FROM CHATS chat " +
```

Comment:
same story here. No need to join tables if you select only columns from one. Please, compare performance. If is's the same, then it's OK.

Reply:
But I need to get all the chats in which all the users in list are in. For this I have only users ID's. So I need to use "USERSCHATS" table to get records from "CHATS" table.

Reply:
It's still the same but without JOIN query is a bit more appropriate